float64 velocity
---
bool ack
